Extensions/BBTextLabel/JsExtension.js

```diff
@@ -0,0 +1,385 @@
+ */
+
+//TODO: add base color, base font, base size
+const MultiStyleText = require('./dist/pixi-multistyle-text.umd');
+console.log(MultiStyleText);
+module.exports = {
+  createExtension: function(_, gd) {
+    const extension = new gd.PlatformExtension();
+    extension
+      .setExtensionInformation(
+        'BBText',
+        'BBText',
+        _('Display a BBText object on the scene.'),
+        'Todor Imreorov',
+        'Open source (MIT License)'
+      )
+      .setExtensionHelpPath('/objects/BBText');
+
+    var ObjectBBText = new gd.ObjectJsImplementation();
+    ObjectBBText.updateProperty = function(
+      objectContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === 'Text') {
+        objectContent.text = newValue;
+        return true;
+      }
+      if (propertyName === 'Color') {
+        objectContent.color = newValue;
+        console.log('base col:', newValue);
+        return true;
+      }
+      if (propertyName === 'Opacity') {
+        objectContent.opacity = parseFloat(newValue);
+        return true;
+      }
+      if (propertyName === 'Visible') {
+        objectContent.visible = newValue === '1';
+        return true;
+      }
+
+      return false;
+    };
+    ObjectBBText.getProperties = function(objectContent) {
+      var objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties.set(
+        'Text',
+        new gd.PropertyDescriptor(objectContent.text)
+          .setType('string')
+          .setLabel(_('bbcode'))
+      );
+
+      objectProperties.set(
+        'Color',
+        new gd.PropertyDescriptor(objectContent.color)
+          .setType('color')
+          .setLabel(_('Base color'))
+      );
+
+      objectProperties.set(
+        'Opacity',
+        new gd.PropertyDescriptor(objectContent.opacity.toString())
+          .setType('number')
+          .setLabel(_('opacity (0-255)'))
+      );
+      objectProperties.set(
+        'Visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    ObjectBBText.setRawJSONContent(
+      JSON.stringify({
+        text: 'insert text',
+        opacity: 255,
+        visible: true,
+      })
+    );
+
+    ObjectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    ObjectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _('Displays a BBcode rich Text.'),
```

Comment:
nitpicking: careful of your casing when writing, "text" should be lowercase (unless it's an acronym in which case you would have capitalized "rich" too)

Reply:
Also: BBCode, not BBcode ;)